common/djangoapps/xblock_django/models.py

```diff
@@ -72,3 +76,125 @@ def __unicode__(self):
             disabled_xblocks=config.disabled_blocks,
             disabled_create_block_types=config.disabled_create_block_types
         )
+
+
+class XBlockConfigFlag(ConfigurationModel):
+    """
+    Enables site-wide configuration for xblock support state.
+    """
+
+    class Meta(object):
+        app_label = "xblock_django"
+
+    def __unicode__(self):
+        return "[XBlockConfigFlag] enabled={}".format(self.enabled)
+
+
+class XBlockConfig(models.Model):
+    """
+    Configuration for a specific xblock. Currently used for support state.
+    """
+    FULL_SUPPORT = 'fs'
+    PROVISIONAL_SUPPORT = 'ps'
+    UNSUPPORTED_OPT_IN = 'ua'
+    UNSUPPORTED_NO_OPT_IN = 'ud'
+    DISABLED = 'da'
+
+    SUPPORT_CHOICES = (
+        (FULL_SUPPORT, _('Fully Supported')),
+        # May lack robustness, course staff should test.
+        (PROVISIONAL_SUPPORT, _('Provisionally Supported')),
+        # Unsupported, may not meet edX standards.
+        (UNSUPPORTED_OPT_IN, _('Unsupported (Opt-in allowed)')),
+        # Used when deprecating components, never allowed to create in Studio.
+        (UNSUPPORTED_NO_OPT_IN, _('Unsupported (Opt-in disallowed)')),
+        # Will not render in the LMS
+        (DISABLED, _('Disabled')),
+    )
+
+    # for archiving
+    history = HistoricalRecords()
+    change_date = models.DateTimeField(auto_now=True, verbose_name=_("change date"))
+    changed_by = models.ForeignKey(
+        User,
+        editable=False,
+        null=True,
+        on_delete=models.PROTECT,
+        # Translators: this label indicates the name of the user who made this change:
+        verbose_name=_("changed by"),
+    )
+
+    name = models.CharField(max_length=255, null=False)
+    template = models.CharField(max_length=255, blank=True, default='')
+    support_level = models.CharField(max_length=2, choices=SUPPORT_CHOICES, default=UNSUPPORTED_NO_OPT_IN)
+    deprecated = models.BooleanField(
+        default=False,
+        verbose_name=_('show deprecation messaging in Studio'),
+        help_text=_(
+            "Only XBlocks listed in a course's Advanced Module List can be flagged as deprecated. "
+            "Note that deprecation is by XBlock name, and is not specific to template.")
```

Comment:
Yeah, I think in that case, I'd probably have a separate model that controls information that is per-XBlock, separate from the template configuration. It might make sense (if you're doing that), to have a 1-many field connecting the XBlock and the template, rather than repeating the XBlock type name in both models.